sdk-api-src/content/d3dcommon/nn-d3dcommon-id3ddestructionotifier.md

```diff
@@ -0,0 +1,108 @@
+---
+UID: NN:d3dcommon.ID3DDestructionNotifier
+title: ID3DDestructionNotifier (d3dcommon.h)
+description: ID3DDestructionNotifier is an include interface that the user implements to allow an application to call user-overridable methods for opening and closing shader
+helpviewer_keywords: ["ID3DDestructionNotifier","ID3DDestructionNotifier interface [Direct3D]","ID3DDestructionNotifier interface [Direct3D]","described","d3dcommon/ID3DDestructionNotifier","direct3d.id3ddestructionnotifier"]
+old-location: direct3d11\id3dinclude.htm
+tech.root: direct3d11
+ms.assetid: 2020ce65-3a6e-4a9f-9e97-b94e3c75f4f5
+ms.date: 12/05/2018
+ms.keywords: ID3DDestructionNotifier, ID3DDestructionNotifier interface [Direct3D], ID3DDestructionNotifier interface [Direct3D],described, d3dcommon/ID3DDestructionNotifier, direct3d.id3ddestructionnotifier
+f1_keywords:
+- d3dcommon/ID3DDestructionNotifier
+dev_langs:
+- c++
+req.header: d3dcommon.h
+req.include-header: 
+req.target-type: Windows
+req.target-min-winverclnt: Windows 7 [desktop apps \| UWP apps]
+req.target-min-winversvr: Windows Server 2008 R2 [desktop apps \| UWP apps]
+req.kmdf-ver: 
+req.umdf-ver: 
+req.ddi-compliance: 
+req.unicode-ansi: 
+req.idl: 
+req.max-support: 
+req.namespace: 
+req.assembly: 
+req.type-library: 
+req.irql: 
+topic_type:
+- APIRef
+- kbSyntax
+api_type:
+- COM
+api_location:
+- D3DCommon.h
+api_name:
+- ID3DDestructionNotifier
+targetos: Windows
+req.typenames: 
+req.redist: 
+ms.custom: 19H1
+---
+
+# ID3DInclude interface
+
+
+## -description
+
+
+<b>ID3DDestructionNotifier</b> is an interface that can be used to register for callbacks when a D3D Nano-COM object is destroyed.
+To acquire an instance of this interface, call <a href="windows/win32/api/unknwn/nf-unknwn-iunknown-queryinterface(refiid_void)"></a> on a D3D object with the **IID** of **ID3DDestructionNotifier**.
+
+
+## -inheritance
+
+The <b xmlns:loc="http://microsoft.com/wdcml/l10n">ID3DDestructionNotifier</b> interface inherits from the <a href="https://docs.microsoft.com/windows/desktop/api/unknwn/nn-unknwn-iunknown">IUnknown</a> interface. <b>ID3DInclude</b> also has these types of members:
```

Comment:
Typo - ID3DInclude?

Reply:
Oops, copied the structure of this from that (as I have no idea what I am doing wrt writing docs). Will fix